File: Easy/1137_NthTribonacciNumber/Java/Solution.java
class Solution {
    public int tribonacci(int n) {
        switch (n) {
            case 0:
                return 0;
            case 1:
                return 1;
            default:
                int t0 = 0, t1 = 1, t2 = 1, t = 1;
                for (int i = 3; i <= n; i++) {
                    t = t0 + t1 + t2;
                    t0 = t1;
                    t1 = t2;
                    t2 = t;
                }
                return t;
        }
    }
}

File: Easy/1137_NthTribonacciNumber/Python3/Solution.py
class Solution:
    def tribonacci(self, n: int) -> int:
        if n == 0: return 0
        if n in [1, 2]: return 1
        memo = [0, 1, 1]
        for i in range(3, n + 1):
            s = sum(memo)
            memo[0] = memo[1]
            memo[1] = memo[2]
            memo[2] = s
        return memo[2]